Lib/_pyrepl/fancycompleter.py

```diff
@@ -0,0 +1,205 @@
+#
+#                        All Rights Reserved
+"""
+Colorful TAB completion for Python prompt
+"""
+from _pyrepl import readline
+from _colorize import ANSIColors
+import rlcompleter
+import types
+from itertools import count
+
+
+class DefaultConfig:
+
+    consider_getitems = True
+    use_colors = 'auto'
+
+    color_by_type = {
+        types.BuiltinMethodType: ANSIColors.BOLD_TEAL,
+        types.MethodType: ANSIColors.BOLD_TEAL,
+        type((42).__add__): ANSIColors.BOLD_TEAL,
+        type(int.__add__): ANSIColors.BOLD_TEAL,
+        type(str.replace): ANSIColors.BOLD_TEAL,
+
+        types.FunctionType: ANSIColors.BOLD_BLUE,
+        types.BuiltinFunctionType: ANSIColors.BOLD_BLUE,
+
+        type: ANSIColors.BOLD_MAGENTA,
+
+        types.ModuleType: ANSIColors.TEAL,
+        type(None): ANSIColors.GREY,
+        str: ANSIColors.BOLD_GREEN,
+        bytes: ANSIColors.BOLD_GREEN,
+        int: ANSIColors.BOLD_YELLOW,
+        float: ANSIColors.BOLD_YELLOW,
+        complex: ANSIColors.BOLD_YELLOW,
+        bool: ANSIColors.BOLD_YELLOW,
+    }
+
+    def setup(self):
+        import _colorize
+        if self.use_colors == 'auto':
+            colors = _colorize.get_colors()
+            self.use_colors = colors.RED != ""
+
+
+class Completer(rlcompleter.Completer):
+    """
+    When doing someting like a.b.<TAB>, display only the attributes of
+    b instead of the full a.b.attr string.
+
+    Optionally, display the various completions in different colors
+    depending on the type.
+    """
+    def __init__(self, namespace=None, Config=DefaultConfig):
+        rlcompleter.Completer.__init__(self, namespace)
+        self.config = Config()
+        self.config.setup()
+
+        # XXX: double check what happens in this case once fancycompleter works
+        if False and hasattr(readline, '_setup'):
+            # this is needed to offer pyrepl a better chance to patch
+            # raw_input. Usually, it does at import time, but is we are under
+            # pytest with output captured, at import time we don't have a
+            # terminal and thus the raw_input hook is not installed
+            readline._setup()
+
+        if self.config.use_colors:
+            readline.parse_and_bind('set dont-escape-ctrl-chars on')
+        if self.config.consider_getitems:
+            delims = readline.get_completer_delims()
+            delims = delims.replace('[', '')
+            delims = delims.replace(']', '')
+            readline.set_completer_delims(delims)
+
+    def complete(self, text, state):
+        """
+        stolen from:
+        http://aspn.activestate.com/ASPN/Cookbook/Python/Recipe/496812
+        """
+        if text == "":
+            return ['\t', None][state]
+        else:
+            return rlcompleter.Completer.complete(self, text, state)
+
+    def _callable_postfix(self, val, word):
+        # disable automatic insertion of '(' for global callables:
+        # this method exists only in Python 2.6+
```

Comment:
Doesn't matter in CPython 3.14 in 2025 :)
```suggestion
        # disable automatic insertion of '(' for global callables
```

Reply:
if anything, this is a proof that fancycompleter has been battle tested for like ~18 years by now :)